knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -125,22 +125,22 @@ def fetch_google_OD(download_link):
                 print("Empty value generated")
             return [nt_time, nt_dist]
 
-        O_gdf = left_input.to_pandas().rename(columns={self.O_geo_col: "geometry"})
-        D_gdf = right_input.to_pandas().rename(columns={self.D_geo_col: "geometry"})
-        O_gdf = gp.GeoDataFrame(O_gdf, geometry="geometry")
-        D_gdf = gp.GeoDataFrame(D_gdf, geometry="geometry")
+        O_gdf = knut.load_geo_data_frame(left_input, self.O_geo_col, exec_context)
+        D_gdf = knut.load_geo_data_frame(right_input, self.D_geo_col, exec_context)
+        O_gdf = O_gdf[[self.O_geo_col]].rename(columns={self.O_geo_col: "geometry"})
+        D_gdf = D_gdf[[self.D_geo_col]].rename(columns={self.D_geo_col: "geometry"})
 
         # Set a lat\Lon CRS
         O_gdf = O_gdf.to_crs(4326)
         D_gdf = D_gdf.to_crs(4326)
         # Generate ID
-        O_gdf["OriginID"] = range(1, (O_gdf.shape[0] + 1))
-        D_gdf["DestinationID"] = range(1, (D_gdf.shape[0] + 1))
+        O_gdf["originid"] = range(1, (O_gdf.shape[0] + 1))
+        D_gdf["destinationid"] = range(1, (D_gdf.shape[0] + 1))
         mergedf = O_gdf.merge(D_gdf, how="cross")
         mergedf_x = gp.GeoDataFrame(geometry=mergedf["geometry_x"])
```

Comment:
As the naming of  Merge function is a little bit complicated, I create new data and rename the geometry column to avoid potential conflict with that: 
       # Create new data
        O_gdf = Or_gdf[[self.O_geo_col]].rename(columns={self.O_geo_col: "geometry"})
        D_gdf = De_gdf[[self.D_geo_col]].rename(columns={self.D_geo_col: "geometry"})

also I build  ID column checking
       # Cross join Data
        Or_gdf = knut.load_geo_data_frame(left_input, self.O_geo_col, exec_context)
        De_gdf = knut.load_geo_data_frame(right_input, self.D_geo_col, exec_context)
        Or_gdf =knut.validify_id_column(Or_gdf,self.O_id_col)
        De_gdf =knut.validify_id_column(De_gdf,self.D_id_col)
        # Create new data and rename
        O_gdf = Or_gdf[[self.O_geo_col]].rename(columns={self.O_geo_col: "geometry"})
        D_gdf = De_gdf[[self.D_geo_col]].rename(columns={self.D_geo_col: "geometry"})

        # Set a lat\Lon CRS
        O_gdf = O_gdf.to_crs(4326)
        D_gdf = D_gdf.to_crs(4326)
        # Generate ID
        O_gdf[self._OID] = Or_gdf[self.O_id_col].to_list()
        D_gdf[self._DID] = De_gdf[self.D_id_col].to_list()

Reply:
Merging data have different naming rules, if  A and B are different, the column names will remain, if they are the same, e.g., A and A then  _X and _Y will be appended to them as  A_X and A_Y. We don't know what geo column name will be input. The second reason is some of the following functions also use "geometry" as default. So I just renaming them t o"geometry"